ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/resources/streaming/WSQueryEndpoint.java

```diff
@@ -264,15 +264,21 @@ public void onError(final Session session, final Throwable t) {
     SessionUtil.closeSilently(session, CloseCodes.UNEXPECTED_CONDITION, t.getMessage());
   }
 
-  private void checkEndpointAuthorization(final Principal userPrincipal, final String methodName) {
-    final Class<?> className = this.getClass();
-
-    if (!securityExtension.getAuthorizer().hasAccess(userPrincipal, className, methodName)) {
-      final String userName = (userPrincipal != null) ? userPrincipal.getName() : null;
-      throw new KsqlException(
-          String.format("User:%s is denied to access this cluster.", userName)
-      );
-    }
+  private void checkAuthorization(final Session session) {
+    final String method = "POST";
+    final String path = this.getClass().getAnnotation(ServerEndpoint.class).value();
+    final Principal user = session.getUserPrincipal();
+
+    securityExtension.getAuthorizationProvider().ifPresent(
+        provider -> {
+          try {
+            provider.checkEndpointAccess(user, method, path);
+          } catch (final Throwable t) {
+            log.warn(String.format("User:%s is denied access \"%s %s\"", user, method, path), t);
```

Comment:
Will this error message be confusing to users of the websocket endpoint? I'm no websocket expert but I think talking about POST-ing to a websocket endpoint is strange.

Reply:
Right. I changed it.